RadixWallet/Clients/ProfileStore/ProfileStore.swift

```diff
@@ -1,284 +1,287 @@
+		@Dependency(\.device) var device
+		if !profile.hasMainnetAccounts() {
+			logAssertionFailure("Incorrect implementation should have accounts on mainnet after finishing onboarding.")
+		}
+		let model = await device.model
+		let name = await device.name
+		let deviceDescription = DeviceInfo.deviceDescription(
+			name: name,
+			model: model
+		)
+		deviceInfo.description = deviceDescription
+		let lastUsedOnDevice = deviceInfo
+		try? secureStorageClient.saveDeviceInfo(lastUsedOnDevice)
+		try? await updating {
+			$0.header.lastUsedOnDevice = lastUsedOnDevice
+			$0.header.creatingDevice.description = deviceDescription
+		}
+	}
 
-		do {
-			// Load the snapshot, also this will validate if the snapshot actually exist
-			let profileSnapshot = try await secureStorageClient.loadProfileSnapshot(header.id)
-			guard let profileSnapshot else {
-				struct FailedToLoadProfile: Swift.Error {}
-				throw FailedToLoadProfile()
+	public func unlockedApp() async -> Profile {
+		loggerGlobal.notice("Unlocking app")
+		let buffered = bufferedOwnershipConflictWhileAppLocked
+		self.mode = .appIsUnlocked
+		if let buffered {
+			loggerGlobal.notice("We had a buffered Profile ownership conflict, emitting it now.")
+			do {
+				try await doEmit(conflictingOwners: buffered)
+				return profile // might be a new one! if user selected "delete"
+			} catch {
+				logAssertionFailure("Failure during Profile ownership resolution, error: \(error)")
+				// Not import enough to prevent app from being used
+				return profile
 			}
-			try await changeProfileSnapshot(to: profileSnapshot)
-		} catch {
-			let errorMessage = "Critical failure, unable to save imported profile snapshot: \(String(describing: error))"
-			loggerGlobal.critical(.init(stringLiteral: errorMessage))
-			assertionFailure(errorMessage) // for DEBUG builds we want to crash
-			throw error
+		} else {
+			return profile
 		}
 	}
+}
 
-	public func commitEphemeral() async throws {
-		let ephemeral = try assertProfileStateIsEphemeral()
-		try await changeProfileSnapshot(to: ephemeral.profile.snapshot())
+extension DeviceInfo {
+	public static func deviceDescription(
+		name: String,
+		model: String
+	) -> String {
+		"\(model) (\(name))"
 	}
+}
 
-	/// If persisted: updates the in-memory across-the-app-used Profile and also
-	/// syncs the new value to secure storage (and iCloud if enabled/able).
-	///
-	/// if ephemeral: Updates the ephemeral profile.
-	public func update(profile: Profile) async throws {
-		guard profile != profileStateSubject.value.profile else {
+// MARK: - ProfileStore.Error
+extension ProfileStore {
+	public enum Error: String, Swift.Error, Equatable {
+		case profileIDMismatch
+		case profileUsedOnAnotherDevice
+	}
+}
+
+// MARK: "Private"
+extension ProfileStore {
+	func _lens<Property>(
+		_ transform: @escaping @Sendable (Profile) -> Property?
+	) -> AnyAsyncSequence<Property> where Property: Sendable & Equatable {
+		profileSubject.compactMap(transform)
```

Comment:
I know this probably old code, but this will update all previous profiles as well, right? If so, what for?

Reply:
old profiles? no? there is only one, `CurrentValueSubject` right?